src/oracles/CoinbaseOracle.sol

```diff
@@ -0,0 +1,120 @@
+    address immutable reporter;
+
+    uint8 public immutable decimals;
+
+    string public symbol;
+
+    uint80 public lastRoundId;
+
+    mapping(uint80 => int256) public roundAnswers;
+    mapping(uint80 => uint256) public roundTimestamps;
+
+    /**
+     * @notice Emitted when a new price is pushed.
+     * @param roundId The round ID of the new price.
+     * @param price The price that was pushed.
+     * @param timestamp The timestamp at which the price was pushed.
+     */
+    event PricePushed(uint80 indexed roundId, int256 price, uint256 timestamp);
+
+    /**
+     * @notice Constructor to initialize the CoinbaseOracle contract.
+     * @param _symbol The symbol of the asset being reported.
+     * @param _decimals The number of decimals in the reported price.
+     * @param _reporter The address of the reporter allowed to push price data.
+     */
+    constructor(string memory _symbol, uint8 _decimals, address _reporter) {
+        symbol = _symbol;
+        decimals = _decimals;
+        reporter = _reporter;
+    }
+
+    /**
+     * @notice Returns the latest round data.
+     * @return roundId The ID of the latest round.
+     * @return answer The latest price.
+     * @return startedAt The timestamp when the round started.
+     * @return updatedAt The timestamp when the round was updated.
+     * @return answeredInRound The round ID in which the answer was computed.
+     */
+    function latestRoundData()
+        external
+        view
+        override
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        int256 latestAnswer = roundAnswers[lastRoundId];
+        uint256 latestTimestamp = roundTimestamps[lastRoundId];
+        return (lastRoundId, latestAnswer, latestTimestamp, latestTimestamp, lastRoundId);
+    }
+
+    /**
+     * @notice Returns the data for a specific round.
+     * @param _roundId The round ID to retrieve the data for.
+     * @return roundId The ID of the round.
+     * @return answer The price of the round.
+     * @return startedAt The timestamp when the round started.
+     * @return updatedAt The timestamp when the round was updated.
+     * @return answeredInRound The round ID in which the answer was computed.
+     */
+    function getRoundData(uint80 _roundId)
+        external
+        view
+        override
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        int256 latestAnswer = roundAnswers[_roundId];
+        uint256 latestTimestamp = roundTimestamps[_roundId];
+        return (_roundId, latestAnswer, latestTimestamp, latestTimestamp, _roundId);
+    }
+
+    /**
+     * @notice Pushes a new price to the oracle.
+     * @param priceData The encoded price data.
+     * @param signature The signature to verify the authenticity of the data.
+     */
+    function pushPrice(bytes memory priceData, bytes memory signature) external {
```

Comment:
There is some inconsistency with some of the formatting in this PR. The function props here do not lead with an `_` but in other functions they do. It actually looks like we are pretty inconsistent in a number of other places as well. we can ignore it in this PR and update the whole repo together in a follow on.